src/main/java/io/trino/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -447,11 +462,120 @@ private void setRanges(TileDBSplit split) throws TileDBError
+                    }
+                    else {
+                        cond = new QueryCondition(ctx, att.getName(), " ".getBytes(), att.getType().javaClass(), TILEDB_EQ);
+                    }
+                    query.setCondition(cond);
+                    continue;
+                }
+                if ((attBounds.getFirst() != null || attBounds.getSecond() != null)) {
+                    if (attBounds.getFirst() == null) {
+                        if (isVar) {
+                            highBound = attBounds.getSecond().toString().getBytes();
+                        }
+                        else {
+                            highBound = attBounds.getSecond();
+                        }
+                        QueryCondition cond = new QueryCondition(ctx, att.getName(), highBound, att.getType().javaClass(), TILEDB_LE);
+                        query.setCondition(cond);
+                    }
+                    else if (attBounds.getSecond() == null) {
+                        if (isVar) {
+                            lowBound = attBounds.getFirst().toString().getBytes();
+                        }
+                        else {
+                            lowBound = attBounds.getFirst();
+                        }
+                        QueryCondition cond = new QueryCondition(ctx, att.getName(), lowBound, att.getType().javaClass(), TILEDB_GE);
+                        query.setCondition(cond);
+                    }
+                    else {
+                        if (isVar) {
+                            highBound = attBounds.getSecond().toString().getBytes();
+                            lowBound = attBounds.getFirst().toString().getBytes();
+                        }
+                        else {
+                            highBound = attBounds.getSecond();
+                            lowBound = attBounds.getFirst();
+                        }
+                        QueryCondition cond1 = new QueryCondition(ctx, att.getName(), lowBound, att.getType().javaClass(), TILEDB_GE);
+                        QueryCondition cond2 = new QueryCondition(ctx, att.getName(), highBound, att.getType().javaClass(), TILEDB_LE);
+                        query.setCondition(cond1.combine(cond2, TILEDB_AND));
+                    }
+                }
+                LOG.info("Query %s setting range for attribute %s to [%s, %s]", queryId, att.getName(), attBounds.getFirst(), attBounds.getSecond());
+            }
+        }
+        catch (Exception ex) {
+            ex.printStackTrace();
         }
-
         recordFunctionTime("setRanges", timer);
     }
 
+    /**
+     * Compute a (lower, upper) bound for the given attribute based on the given split.
+     */
+    private Pair getBoundsForAttribute(TileDBSplit split, Attribute attribute) throws TileDBError
+    {
+        if (emptyQueryAttributes.contains(attribute.toString())) {
```

Comment:
This, emptyQueryAttributes keeps all the attributes where a query of type "SELECT * FROM TABLE WHERE name="". The empty string would normally create a range [ , ]. However when dealing with multiple splits, some of them would return [null, null] which return the whole domain and produces wrong results. I am not sure why this happens though.